### PR TITLE
Fallback data provider for balances

### PR DESCRIPTION
**Pull request by brunobar79**
This PR implements a fallback provider based on:
- Etherscan: for token transfer tx history  - So we know what tokens are in the wallet
- Coingecko: for current prices
- Onchain multibalance contract via Infura

The fallback provider will kick in automatically when we don't get balances / prices from zerion.
Those scenarios are:
a) Zerion returning an empty array of assets.
b) Not hearing back from zerion after 10 seconds after starting to listen for an address

Zerion connection is still active, so if we hear back in the future, we'll switch automatically to it and turn off the fallback data provider.

Also made pull to refresh in the WalletScreen to reinitialize the explorer (Zerion connection), so if that fails it would automatically fallback to the new data provider.

NOTE: This doesn't include transaction history. If we decide to do that too it can be added in a separate PR.

There are multiple ways to test each part of this:

1) To simulate "didnt hear back from zerion after 10s" go to `src/redux/explorer.js` and comment the block:
```
socket.on(messages.ADDRESS_ASSETS.RECEIVED, message => {
...
})
```

2) To simulate "Zerion returned an empty array of asset"  change the line 151 of `src/redux/explorer.js` from 
`return  true` to `return false`

### CODE REVIEW

**Review by mikedemarais — APPROVED** [review]
cool visually looks good and clean and i cant spot any low hanging stuff! 

@jinchung should be official reviewer tho cuz shes most familiar with this business logic!

**Comment by christianbaroni** [comment]
@brunobar79 @jinchung @mikedemarais is there anything blocking merging this?

**Comment by brunobar79** [comment]
@christianbaroni  This is still waiting for a review from @jinchung

**Comment by osdnk** [comment]
@brunobar79 Can you rebase?

File: src/redux/data.js
Comment (jinchung) [inline src/redux/data.js:241]: These shitcoins and the compound tokens should not be included in assets (given they should be filtered out on line 221 with "compound" and "trash" types - were you still seeing these included?

**Review by skylarbarrera — APPROVED** [review]
LGTM!

A few comments:

We may want to tone down the FallbackExplorer logging, it floods the log with updating assets, balances, & price & slso found new assets.

There is a small lag when switching between Zerion & the fallback, most times it looks as follows, I had one instance where prices/% changed was NaN for maybe half a sec. Don't think this is worth spending any time on right now.

https://cloud.skylarbarrera.com/Screen-Recording-2020-10-13-11-57-31.mp4

**Comment by brunobar79** [comment]
@skylarbarrera  thanks for looking at it!
- For the record, the logging only happens when we fallback (meaning zerion is down or broken) so it shouldn't be a problem.
- The lag is because we had to do a bunch of request the first time (coingecko, etherscan, etc) but it's still better than being broken

**Comment by osdnk** [comment]
I guess if it displays "Nan" for a moment, it should be fixed if it's not a big deal :)

**Review by skylarbarrera — APPROVED** [review]
Cant reproduce, I think we're good! 👍

**Review by jinchung — APPROVED** [review]
🥑

**Pull request merged**
